### ISSUE TITLE
Look for a nightly compiler

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/cargo-expand#10]
```
$ rustup run stable cargo expand >/dev/null

  warning: the option `Z` is unstable and should only be used on the nightly
    compiler, but it is currently accepted for backwards compatibility; this
    will soon change, see issue #31847 for more details
  warning: the option `pretty` is unstable and should only be used on the nightly
    compiler, but it is currently accepted for backwards compatibility; this
    will soon change, see issue #31847 for more details
```

Let's run `rustc --version` and `rustup run nightly rustc --version` and choose the one that provides a nightly compiler.

If neither is nightly, keep the current behavior with warnings.

**Comment by ehuss** [comment]
Note: as of Rust 1.19 you *must* use nightly in order for cargo expand to work.

I'm curious if you have any thoughts about the future of `cargo expand`.    In rust-lang/rust#31847 it is mentioned that `--pretty` is broken and may be removed or replaced.

**Comment by dtolnay** [comment]
Fixed in db455ab7266ee1440ee2b3bd535a548806a89513.
